### DISCUSSION TITLE
[v1.10] How to use LNBits with a node behind Tor

### DISCUSSION THREAD

**Discussion by adevoss** [raspiblitz/raspiblitz#4195] [Q&A]
How LNBits is exposed to internet (clearnet) when my node is behind Tor?
What is the status of ip2tor?
Both shops fail (fulmo not reachable, the other one shows subscriptions but payment fails)

Came across this tutorial:
https://github.com/openoms/bitcoin-tutorials/blob/master/tor2ip_tunnel.md
Is this still current? Then I can host my own.

**Comment by rootzoll** [comment]
@raulcano can you check your ip2tor payment?

**Comment by rootzoll** [comment]
@openoms is this tutorial still up2date? https://github.com/openoms/bitcoin-tutorials/blob/master/tor2ip_tunnel.md

**Comment by raulcano** [comment]
I just did. It's working fine. What seems to be the issue?

rootzoll ***@***.***> schrieb am Mi., 4. Okt. 2023, 11:19:

**Comment by rootzoll** [comment]
> I just did. It's working fine. What seems to be the issue?

@adevoss can you report your payment fails on the IP2Tor shop a bit more in detail?

**Comment by adevoss** [comment]
What do you want to know? Fulmo shop is down. ip2tor.com Shop (Sweden 30 day 2200 sats option) failed twice before.
Didn't try it now.

Testing my own ip2tor bridge (@openoms tutorial). Now fails with a 502. Could be anything. Just a first try.
Also testing with reverse proxy without tor. Without tor LNBits is now reachable from clearnet.

**Comment by rootzoll** [comment]
@raulcano I tested the ip2tor.com on my test node with one open channel - payment failed also .. because the node was not able to find a route. Can you post a node URI that a user can open a direct channel with for better payment success?

**Comment by raulcano** [comment]
Strange... I keep receiving new subscriptions. Thanks for the testing, BTW.
Anyway, this is the node page, with all the info there to connect and more
https://1ml.com/node/033b3c92d971261505b9a6e2e39c6de82a84b97ca6e720ab2809c451a2374b94f5

rootzoll ***@***.***> schrieb am Mi., 18. Okt. 2023, 10:45:

**Comment by rootzoll** [comment]
@adevoss I can confirm also with my test node - when I open a direct channel with https://1ml.com/node/033b3c92d971261505b9a6e2e39c6de82a84b97ca6e720ab2809c451a2374b94f5 then the payment works with ip2tor.com

@raulcano can you add into the description of your services in the ip2tor shop the info: If you payment fails, try to open a direct channel with node 033b3c92d971261505b9a6e2e39c6de82a84b97ca6e720ab2809c451a2374b94f5"? I would the add in case of a failed payment the info that the user should check the description for a nodeid to try a direct channel with. see #4228

**Reply by raulcano** [reply]
I updated the field with the ToS for each host. Can you try opening the hosts to see if the ToS description includes now the node's public key?
(The description field does not allow that many characters and it will take me a while to update the whole thing to allow that)

**Comment by raulcano** [comment]
Thanks for the tests, @rootzoll.
This evening, I'll update the descriptions on the hosts as you suggested.
I'm nevertheless surprised that my channel, which is connected to a Voltage
node, is not enough to be reachable. One would think that Voltage nodes are
pretty much reachable by anyone...

rootzoll ***@***.***> schrieb am Di., 24. Okt. 2023, 11:47:

**Reply by openoms** [reply]
It is possible that the "route not found" error is because the high routing fee set on the connected node.

Try to set the max fee higher when paying.

**Reply by raulcano** [reply]
Is this something I need to do as the owner of the LN node that receives the payment? Or do you mean to address the buyers of ip2tor bridges?

**Reply by openoms** [reply]
The routing fee is paid by the payer. There is nothing to set in the destination, but it can be solved with incoming channels (capacity) with well connected nodes charging less.

With a few new channels now it should be probably fine: https://amboss.space/node/033b3c92d971261505b9a6e2e39c6de82a84b97ca6e720ab2809c451a2374b94f5?section=Channels

**Comment by raulcano** [comment]
Ok, thanks, I figured that would be the case.
I need to research then a bit more about connecting to other nodes to
mitigate that issue.
Cheers!

openoms ***@***.***> schrieb am Mi., 25. Okt. 2023, 08:03:

**Comment by rootzoll** [comment]
OK I created a PR #4246 to address this. Closing issue.
